src/components/ha-color-picker.ts

```diff
@@ -33,137 +31,177 @@ export class HaColorPicker extends LitElement {
 
   @property({ type: Boolean }) public disabled = false;
 
-  @query("ha-select") private _select?: HaSelect;
-
-  connectedCallback(): void {
-    super.connectedCallback();
-    // Refresh layout options when the field is connected to the DOM to ensure current value displayed
-    this._select?.layoutOptions();
-  }
-
-  private _valueSelected(ev) {
-    ev.stopPropagation();
-    if (!this.isConnected) return;
-    const value = ev.target.value;
-    this.value = value === this.defaultColor ? undefined : value;
-    fireEvent(this, "value-changed", {
-      value: this.value,
-    });
-  }
+  @property({ type: Boolean }) public required = false;
 
   render() {
-    const value = this.value || this.defaultColor || "";
-
-    const isCustom = !(
-      THEME_COLORS.has(value) ||
-      value === "none" ||
-      value === "state"
-    );
+    const effectiveValue = this.value ?? this.defaultColor ?? "";
 
     return html`
-      <ha-select
-        .icon=${Boolean(value)}
+      <ha-generic-picker
+        .hass=${this.hass}
+        .disabled=${this.disabled}
+        .required=${this.required}
```

Comment:
```suggestion
        .required=${this.required}
        .hideClearIcon=${!this.value && !!this.defaultColor}
```

Otherwise you could clear the default selection

Reply:
Good point, this works